python/cudf/cudf/core/dataframe.py

```diff
@@ -6454,13 +6487,51 @@ def _apply_support_method(self, method, axis=0, *args, **kwargs):
                 "support `bool_only`."
                 raise NotImplementedError(msg)
 
-            prepared = self._prepare_for_rowwise_op()
+            prepared, mask, common_dtype = self._prepare_for_rowwise_op(
+                method, skipna
+            )
+            for col in prepared._data.names:
+                if prepared._data[col].nullable:
+                    prepared._data[col] = (
+                        prepared._data[col]
+                        .astype(
+                            cudf.utils.dtypes.get_min_float_dtype(
+                                prepared._data[col]
+                            )
+                        )
+                        .fillna(np.nan)
+                    )
             arr = cupy.asarray(prepared.as_gpu_matrix())
 
-            result = getattr(arr, method)(axis=1, **kwargs)
+            if skipna is not False and method in _cupy_nan_methods_map:
+                method = _cupy_nan_methods_map[method]
+
+            result = getattr(cupy, method)(arr, axis=1, **kwargs)
 
             if len(result.shape) == 1:
```

Comment:
Can we use `result.ndim` here instead to be more explicit in what we're checking?